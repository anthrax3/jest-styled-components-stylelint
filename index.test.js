require('./index')({ failOnError: false })
const React = require('react')
const styled = require('styled-components').default
const TestRenderer = require('react-test-renderer')

test('causes styles with lint to log errors', () => {
  // prettier-ignore
  const Title = styled.h1`
    color: red;
    line-height: -
  `
  const wrapper = TestRenderer.create(
    <Title>Who wrote these styles anyway??</Title>
  )
  expect(wrapper.toJSON()).toMatchSnapshot()
})

test('styles without lint log nothing', () => {
  // prettier-ignore
  const Title = styled.h1`
    color: red;
  `
  const wrapper = TestRenderer.create(
    <Title>Who wrote these styles anyway??</Title>
  )
  expect(wrapper.toJSON()).toMatchSnapshot()
})

test('passes stylelint', () => {
  const Title = styled.h1`
    color: red;
  `
  TestRenderer.create(<Title />)
})

test('does not pass stylelint', () => {
  const Title = styled.h1`
    color: {oops!}
  `
  TestRenderer.create(<Title />)
})
