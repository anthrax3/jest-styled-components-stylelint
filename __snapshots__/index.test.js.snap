// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`causes styles with lint to log errors 1`] = `
<h1
  className="indextest__Title-anlvrb-0 ImrkY"
>
  Who wrote these styles anyway??
</h1>
`;

exports[`styles without lint log nothing 1`] = `
<h1
  className="indextest__Title-anlvrb-1 betQCg"
>
  Who wrote these styles anyway??
</h1>
`;
